src/frontend/magnify/src/i18n/TranslationProvider/TranslationsProvider.tsx

```diff
@@ -9,31 +12,88 @@ export interface TranslationProviderProps {
   /**
    * The current locale
    */
-  locale: string;
+  locale?: string;
   /**
    * The default locale
    */
   defaultLocale: string;
-  /**
-   * Messages for the current locale
-   *
-   * @see loadLocaleData to fetch messages for a given locale for this lib
-   * You may need to merge it with your own project messages
-   */
-  messages: Record<string, string> | Record<string, MessageFormatElement[]> | undefined;
+
+  initTranslation?: boolean;
+}
+
+async function getTranslation(
+  locale: string,
+): Promise<Maybe<Record<string, string> | Record<string, MessageFormatElement[]>>> {
```

Comment:
Seems `Maybe` is wrong here as you default translatedMessages to `{}` in case of error.
```suggestion
): Promise<Record<string, string> | Record<string, MessageFormatElement[]>> {
```

Reply:
Agreed